examples/BasicOperations/GetResponsiveSearchAds.php

```
@@ -0,0 +1,175 @@
+                    "\t%s: %s%s",
+                    $error->getErrorCode()->getErrorCode(),
+                    $error->getMessage(),
+                    PHP_EOL
+                );
+            }
+        } catch (ApiException $apiException) {
+            printf(
+                "ApiException was thrown with message '%s'.%s",
+                $apiException->getMessage(),
+                PHP_EOL
+            );
+        }
+    }
+
+    /**
+     * Runs the example.
+     *
+     * @param GoogleAdsClient $googleAdsClient the Google Ads API client
+     * @param int $customerId the customer ID
+     * @param int|null $adGroupId the ad group ID for which responsive search ads will be retrieved.
+     *     If `null`, returns from all ad groups
+     */
+    public static function runExample(
+        GoogleAdsClient $googleAdsClient,
+        int $customerId,
+        ?int $adGroupId
+    ) {
+        $googleAdsServiceClient = $googleAdsClient->getGoogleAdsServiceClient();
+        // Creates a query that retrieves responsive search ads.
+        $query =
+            'SELECT ad_group.id, '
+            . 'ad_group_ad.ad.id, '
+            . 'ad_group_ad.ad.responsive_search_ad.headlines, '
+            . 'ad_group_ad.ad.responsive_search_ad.descriptions, '
+            . 'ad_group_ad.status '
+            . 'FROM ad_group_ad '
+            . 'WHERE ad_group_ad.ad.type = RESPONSIVE_SEARCH_AD '
+            . 'AND ad_group_ad.status != "REMOVED"';
+        if (!is_null($adGroupId)) {
+            $query .= " AND ad_group.id = $adGroupId";
+        }
+
+        // Issues a search request by specifying page size.
+        $response =
+            $googleAdsServiceClient->search($customerId, $query, ['pageSize' => self::PAGE_SIZE]);
+
+        // Iterates over all rows in all pages and prints the requested field values for
+        // the responsive search ad in each row.
+        foreach ($response->iterateAllElements() as $googleAdsRow) {
```

If no rows are returned, it will fail silently. I think a useful message would be helpful and a good practice.

---

Done.